components/clarinet-cli/src/lsp/native_bridge.rs

```diff
@@ -181,6 +182,21 @@ impl LanguageServer for LspNativeBridge {
         Ok(None)
     }
 
+    async fn signature_help(&self, params: SignatureHelpParams) -> Result<Option<SignatureHelp>> {
+        let _ = match self.request_tx.lock() {
+            Ok(tx) => tx.send(LspRequest::SignatureHelp(params)),
+            Err(_) => return Ok(None),
+        };
+
+        let response_rx = self.response_rx.lock().expect("failed to lock response_rx");
+        let ref response = response_rx.recv().expect("failed to get value from recv");
```

Comment:
Generally speaking I would not panic if we can't get the lock, just log and return None.

Reply:
I'll do the changes in the PR I'm currently working on (and for all the requests) @lgalabru